web/src/stream_ui_updates.js

```diff
@@ -107,6 +107,34 @@ export function update_regular_sub_settings(sub) {
     }
 }
 
+// This function updates the state of the default stream and stream privacy settings.
+// If a private stream option is selected, the default stream option is disabled.
+// If the default stream option is checked, the private stream options are disabled.
+export function update_default_stream_and_stream_privacy_state($container) {
+    const $default_stream = $container.find(".default-stream");
+    const $private_stream_labels = $container.find(
+        `label:has(input[value='${CSS.escape(
+            stream_data.stream_privacy_policy_values.private_with_public_history.code,
+        )}']),
+         label:has(input[value='${CSS.escape(
+             stream_data.stream_privacy_policy_values.private.code,
+         )}'])`,
+    );
+
+    const is_default_stream = $default_stream.find("input").prop("checked");
+    const privacy_type = $container.find("input[type=radio][name=privacy]:checked").val();
+    const is_invite_only =
+        privacy_type === "invite-only" || privacy_type === "invite-only-public-history";
+
+    $default_stream
+        .find("#id_is_default_stream_label")
```

Comment:
It feels odd to use IDs in ".find" since IDs are unique and we can instead use them directly. But I guess ID is not used because there are two of them here - one for stream creation and one for stream edit. I think would be better to instead use the class here.